src/main/webapp/scripts/group-script.js

```diff
@@ -202,13 +337,25 @@ async function loadGroups() {
                 if (type == "GROUP") {
                     // there is a nested group inside the current group
                     containsSubGroups = true;
-                    var group = groups[groups.findIndex(elem => elem.id == members[j].id)];
+                    var indexOfGroup = groups.findIndex(elem => elem.id == members[j].id)
+                    var group;
+                    if (indexOfGroup < 0) {
+                        // retrieve group from API
+                        group = await getGroup(members[j].id);
+                    } else {
+                        group = groups[indexOfGroup];
+                    }
                     newCircle.children.push({
                         name: group.name,
                         value: parseInt(group.directMembersCount),
                         id: group.id
                     })
                 } else if (type == "USER") {
+                    var indexOfUser = users.findIndex(elem => elem.id == members[j].id)
+                    if (indexOfUser < 0) {
```

Comment:
add comment noting that indexOfUser will be less than 0 (i believe -1) if it is not present